atlasdb-config/src/main/java/com/palantir/atlasdb/factory/ServiceDiscoveringAtlasSupplier.java

```diff
@@ -64,11 +64,33 @@ public ServiceDiscoveringAtlasSupplier(KeyValueServiceConfig config, Optional<Le
                         "No atlas provider for KeyValueService type " + config.type() + " could be found."
                         + " Have you annotated it with @AutoService(AtlasDbFactory.class)?"
                 ));
-        keyValueService = Suppliers.memoize(() -> atlasFactory.createRawKeyValueService(config, leaderConfig));
+        keyValueService = Suppliers.memoize(() -> pollForKeyValueService(atlasFactory));
         timestampService = () -> atlasFactory.createTimestampService(getKeyValueService());
         timestampStoreInvalidator = () -> atlasFactory.createTimestampStoreInvalidator(getKeyValueService());
     }
 
+    private KeyValueService pollForKeyValueService(AtlasDbFactory atlasFactory) {
+        long checkStartTime = System.currentTimeMillis();
+        while (true) {
+            try {
+                return atlasFactory.createRawKeyValueService(config, leaderConfig);
+            } catch (Exception ex) {
+                if (System.currentTimeMillis() - checkStartTime > config.pollForDataBaseOnStartUpMillis()) {
+                    break;
+                } else {
+                    try {
```

Comment:
Should you not use exponential backoff here to avoid Cassandra being overwhelmed when it comes online?